lib/miner.go

```diff
@@ -198,10 +198,6 @@ func (desoMiner *DeSoMiner) MineAndProcessSingleBlock(threadIndex uint32, mempoo
 		return nil, fmt.Errorf("DeSoMiner._startThread: _mineSingleBlock returned nil; should only happen if we're stopping")
 	}
 
-	if desoMiner.params.IsPoSBlockHeight(blockToMine.Header.Height) {
```

Comment:
The above operations can fail, so adding the PoS block height check here isn't safe. We need to perform the block height check before this point to exit early